File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> v;
        if(head == NULL)
        {
            return true;
        }
        ListNode *curr = head;
        while(curr!=NULL)
        {
            v.push_back(curr->val);
            curr = curr->next;
        }
        vector<int> vt;
        for(int i=0;i<v.size();i++)
        {
            vt.push_back(v[i]);
        }
        reverse(vt.begin(),vt.end());

        if(vt == v)
        {
            return true;
        }
        return false;
    }
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        if(head == null){
            return true;
        }
        ArrayList<Integer> ar = new ArrayList<Integer>();
        ListNode curr = head;

        while(curr!=null){
            ar.add(curr.val);
            curr = curr.next;
        }
        int m = ar.size();
       for(int i = 0; i < m; i++) {
    if(ar.get(i) != ar.get(m - i - 1)) {
        return false;
    }
}
        return true;
    }
}